src/pkg/services/m365/m365.go

```diff
@@ -61,12 +65,87 @@ func UsersCompat(ctx context.Context, acct account.Account) ([]*User, error) {
 func UsersCompatNoInfo(ctx context.Context, acct account.Account) ([]*UserNoInfo, error) {
 	errs := fault.New(true)
 
-	users, err := usersNoInfo(ctx, acct, errs)
+	usersResult, err := usersNoInfo(ctx, acct, errs)
 	if err != nil {
 		return nil, err
 	}
 
-	return users, errs.Failure()
+	return usersResult, errs.Failure()
+}
+
+// UserHasMailbox returns true if the user has any mailbox enabled
+// false otherwise, and a nil pointer and an error in case of error
+func UserHasMailbox(ctx context.Context, acct account.Account, userID string) (*bool, error) {
```

Comment:
avoid pointers on primitive values wherever possible.  Idiomatic golang is to prefer the zero-value of a type whenever possible.  If err is non-nil, the caller should assume the bool to be either false or unusable.
```suggestion
func UserHasMailbox(ctx context.Context, acct account.Account, userID string) (bool, error) {
```